app/controllers/ATSPAYEDataController.scala

```diff
@@ -42,17 +42,25 @@ class ATSPAYEDataController @Inject()(npsService: NpsService, payeAuthAction: Pa
 
   def getATSDataMultipleYears(nino: String, yearFrom: Int, yearTo: Int): Action[AnyContent] = payeAuthAction.async {
     implicit request =>
-      def dataList: Seq[Future[Option[JsValue]]] = (yearFrom to yearTo).toList map { year =>
+      def dataList: Seq[Future[Either[Int, JsValue]]] = (yearFrom to yearTo).toList map { year =>
         callConnector(nino, year) map {
-          case Right(response) => Some(Json.toJson(response))
+          case Right(response) => Right(Json.toJson(response))
           case Left(error) =>
             logger.error(s"Fetching $year data for $nino returned ${error.status}")
-            None
+            Left(error.status)
         }
       }
-      Future.sequence(dataList) map { data =>
-        val flattenedData = data.flatten
-        if (flattenedData.isEmpty) NotFound(s"No data found for $nino") else Ok(Json.toJson(flattenedData))
+
+      Future.sequence(dataList) map { seqEither =>
+        val seqJsValue = seqEither.filter(either => either.isRight) flatMap { right =>
```

Comment:
Previously, we would return a `Seq[Option[JsValue]]` which we then `flatten` which gives us all the years containing data, filtering out the `None` cases. 

Here it looks like. you are trying to collect the Right values, we can simple use `.collect` here and just collect the Right values: 
`seqEither.collect { case Right(value) => value}`

Reply:
done